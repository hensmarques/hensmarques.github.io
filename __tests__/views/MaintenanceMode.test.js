import React from 'react';
import { MaintenanceOverlay } from '../../src/views/MaintenanceOverlay';

const DEFAULT_TEXT = 'This page is currently undergoing maintenance.';
const DEFAULT_TRANSLATE = 'MAINTENANCE_MODE.CONTENT';

describe('<MaintenanceOverlay/>', () => {
  const mockProps = {
    msg: {
      translate: DEFAULT_TRANSLATE,
      msg: DEFAULT_TEXT,
    },
    enabled: true,
  };
  const wrapper = shallow(<MaintenanceOverlay {...mockProps} />);

  it('renders 1 .maintenance-mode inside', () => {
    expect(wrapper.find('.maintenance-mode').length).toBe(1);
  });

  it('renders 1 .maintenance-mode-content inside', () => {
    expect(wrapper.find('.maintenance-mode-content').length).toBe(1);
    expect(wrapper.containsAllMatchingElements([
      <div
        className="maintenance-mode-content"
        dangerouslySetInnerHTML={{
          __html: _t(mockProps.msg.msg, mockProps.msg.translate),
        }}
      />,
    ])).toBe(true);
  });

  it('renders and matches the snapshot', () => {
     expect(wrapper).toMatchSnapshot();
  });
});
