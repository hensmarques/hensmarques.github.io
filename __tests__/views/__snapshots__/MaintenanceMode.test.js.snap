// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaintenanceOverlay/> renders and matches the snapshot 1`] = `
<div
  className="maintenance-mode"
>
  <div
    className="maintenance-mode-container"
  >
    <div
      className="logo-container"
    >
      <img
        className="login-home-splash-logo company-logo-svg"
        src="images/logos/company-logo-white.png"
      />
    </div>
    <div
      className="maintenance-mode-content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "This page is currently undergoing maintenance.",
        }
      }
    />
  </div>
</div>
`;
